### PR TITLE
docs(v9): Migration documents for selected v0 components

### PR DESCRIPTION
**Pull request by jurokapsiar**
Adds migration/upgrade docs for first v0 components:
Avatar, Button, Header, Image, Popup, Tooltip

Adds migration docs for accessibility, positioning and styling

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/3c3bd1fd66dd8d717c3a56cb2204681f9907e64c'>3c3bd1fd66dd8d717c3a56cb2204681f9907e64c</a></sub>
<div id="bundleSizeComment9423"/>

File: apps/public-docsite-v9/src/Concepts/Upgrade/FromV0/Components/Popup.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+```
+
+After:
+
+```tsx
+import { Popover, PopoverTrigger, PopoverSurface, Button } from '@fluentui/react-components';
+const Component = () => (
+  // v0 Popup has default position/align above/start, while v9 has default above/center. Therefore adding `positioning` prop here
+  // v0 Popup no arrow by default, while v9 has. Therefore adding `noArrow` prop here
+  <Popover positioning={'above-start'} noArrow>
```
Comment (spmonahan) [inline apps/public-docsite-v9/src/Concepts/Upgrade/FromV0/Components/Popup.stories.mdx:23]: `Popover` no longer has a `noArrow` prop, [it was changed to `withArrow`](https://github.com/microsoft/fluentui/pull/23205) to align with `Tooltip`.

File: apps/public-docsite-v9/src/Concepts/Upgrade/FromV0/migrate-custom-accessibility.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    <Button role="menuitem" content="First" />
+    ...
+  </Grid>
+);
+```
+
+After:
+
+```tsx
+import { Grid } from '@fluentui/react-components';
```
Comment (spmonahan) [inline apps/public-docsite-v9/src/Concepts/Upgrade/FromV0/migrate-custom-accessibility.stories.mdx:38]: There isn't a `Grid` control in `react-components`. Ideally examples should always use actual components and the code should be more or less copy-pasteable.

File: apps/public-docsite-v9/src/Concepts/Upgrade/FromV0/migrate-positioning.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { Meta } from '@storybook/addon-docs';
+
+<Meta title="Concepts/Upgrading/from v0/Positioning" />
+
+# Mmigrate Positioning Props
```
Comment (spmonahan) [inline apps/public-docsite-v9/src/Concepts/Upgrade/FromV0/migrate-positioning.stories.mdx:5]:
```suggestion
# Migrate Positioning Props
```

File: apps/public-docsite-v9/src/Concepts/Upgrade/FromV0/migrate-positioning.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+In v9, these props are grouped together under `positioning` prop.
+
+## Recommended: positioning shorthand
+
+If only `position` or `align` or both prop is used, for example `<Popup position="below" align="end" />`, it can be migrate to a string like `positioning="below-end"`.
+
+This is v9 positioning shorthand syntax, which avoids using a full blown javascript object. See more on [v9 docsite example](?path=/docs/concepts-developer-positioning-components--default#shorthand-positions).
+
+## migrate other positioning props:
```
Comment (spmonahan) [inline apps/public-docsite-v9/src/Concepts/Upgrade/FromV0/migrate-positioning.stories.mdx:20]:
```suggestion
## Migrate other positioning props:
```

File: apps/public-docsite-v9/src/Concepts/Upgrade/FromV0/Components/Header.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+<Meta title="Concepts/Upgrading/from v0/Components/Header Upgrade" />
+
+# Header Upgrade
+
+## Overview:
+
+`Header` component is removed. Instead, use `Text` or typography components and specify heading in the `as` property. Typography components are based on the `Text` component, they share the same props but comes with added styles.
+
+Typography components:
```
[the remark this answers is not in the document]
Comment (jurokapsiar) [reply apps/public-docsite-v9/src/Concepts/Upgrade/FromV0/Components/Header.stories.mdx:11]: yes, but Text is not clear yet as the v9 Text is very different from v0. For now we do not have a guide for it. it is wip.

File: apps/public-docsite-v9/src/Concepts/Upgrade/FromV0/migrate-custom-accessibility.stories.mdx (hunk 2 above)
Comment (jurokapsiar) [reply apps/public-docsite-v9/src/Concepts/Upgrade/FromV0/migrate-custom-accessibility.stories.mdx:38]: good catch. There is a shim in TMP that is not yet here. will fix.

**Review by ling1726 — APPROVED** [review]

**Pull request merged**